blocks/rich-text/index.js

```diff
@@ -774,15 +778,15 @@ export class RichText extends Component {
 
 	/**
 	 * Calling onSplit means we need to abort the change done by TinyMCE.
-	 * we need to call updateContent to restore the initial content before calling onSplit.
+	 * we need to call setContent to restore the initial content before calling onSplit.
 	 *
 	 * @param {Array}  before content before the split position
 	 * @param {Array}  after  content after the split position
 	 * @param {?Array} blocks blocks to insert at the split position
 	 */
 	restoreContentAndSplit( before, after, blocks = [] ) {
-		this.updateContent();
-		this.props.onSplit( before, after, ...blocks );
+		this.setContent( before );
```

Comment:
Oh I see, so you apply the content directly beforing calling the `onSplit`. I believe we avoided this until now because the block can decide to not split and don't support the `Enter` behavior. If the block don't provide the `onSplit` it's better to keep content as is instead of removing the part that's after the split point. How do you handle this?

Reply:
If there's no `onSplit` prop, `restoreContentAndSplit` will never be called, just like before? `updateContent` is being replaced by `setContent` here, because the only thing we want to do is set the content.

Reply:
There was actually an inconsistency where some use called `setContent` before `restoreContentAndSplit`.

Reply:
👍 Sounds good to me. Let's try to build a good deprecation strategy though

Reply:
Will do :) I'll ponder a bit over maybe using `onReplace`... would be much simpler, but right now `onSplit` is also used as an indicator whether splitting is possible as you mentioned, whereas onReplace is used for other things to. One solution to both problems might be to have something like this:

```
onSplit: ( value ) => createBlock ( name, {
  content: value,
}
```

So here the block provided RichText with the info it needs to split the block, and then we can use `onReplace` with `[ ...blocks, onSplit( value ) ]`.

But one problem with this is that we currently insert an empty paragraph for the list block if `blocks` is empty...

Reply:
Also maybe `onSplit` always create a `defaultBlock` for the "after" content in which case, we can replace it entirely?

Reply:
This need will go away though once we use `innerBlocks` for the list block too...

Reply:
Atm I'm tempted to continue passing the `before` value, but remove the `setAttributes`, then revise the whole function once we work on lists.

Reply:
Fine to leave for another refactor though 👍

Reply:
Cool, let's take care of this when we handle lists.